### PR TITLE
Bugfix/1312537 script stripping disable objects

### PR DESCRIPTION
**Pull request by lopezt-unity**
### Purpose of this PR

Fixing script stripping when the scene has disabled objects and the application is built.
Also fixing script stripping that was not taking the new ShapeComponents into account and creating errors in PB5.0.

### Links

**Fogbugz:** https://fogbugz.unity3d.com/f/cases/1312537/

### Comments to Reviewers

[List known issues, planned work, provide any extra context for your code.]

### CODE REVIEW

**Comment by lopezt-unity** [comment]
@karl- , the problem was notified for PB4.4, should we backport the solution on the PB4.5 branch?

**Review by gytis-zasciurinskas — APPROVED** [review]
We did a peer review with Thomas for this. We have set up a small project to check if the scripts are stripped in the Build and everything is now working as expected. I will attach this project to the FogBugz ticket for future reference (as currently there was no repro project attached)
Also checked the use case where the Shape Renderer would be removed when Shape's scripts were stripped in the Editor (Using Tools -> ProBuilder ->  Action). This is now also working as expected.

File: Editor/EditorCore/StripProBuilderScripts.cs [hunk 1]
```diff
@@ -125,5 +118,16 @@ public static void DoStrip(ProBuilderMesh pb)
...
+
+        static void DestroyProBuilderMeshAndDependencies(GameObject go, ProBuilderMesh pb)
+        {
+            DestroyImmediate(pb);
+
+            if (go.TryGetComponent(out Entity entity))
+                DestroyImmediate(entity);
+
+            if (go.TryGetComponent(out ShapeComponent shapeComponent))
+                DestroyImmediate(shapeComponent);
```
Comment (karl-) [inline Editor/EditorCore/StripProBuilderScripts.cs:130]: is this missing spline and polyshape components?

Comment (karl-) [reply Editor/EditorCore/StripProBuilderScripts.cs:130]: also, can this use the same function as what we have in `UnityScenePostProcessor`?

File: Tests/Editor/Actions/StripProBuilderScriptsTest.cs [hunk 2]
```diff
@@ -45,4 +47,78 @@ public void StripProBuilderScripts_RemovesMeshAndBezierComponents()
...
+        Assume.That(go.GetComponent<ProBuilderMesh>() != null);
+        Assume.That(go.GetComponent<ShapeComponent>() != null);
+
+        StripProBuilderScripts.DoStrip(go.GetComponent<ProBuilderMesh>());
+
+        Assert.That(go.GetComponent<ProBuilderMesh>() == null);
+        Assert.That(go.GetComponent<ShapeComponent>() == null);
+
+        UObject.DestroyImmediate(go);
+    }
```
Comment (karl-) [inline Tests/Editor/Actions/StripProBuilderScriptsTest.cs:67]: awesome to see tests added for this 🚀

File: Editor/EditorCore/StripProBuilderScripts.cs (hunk 1 above)
Comment (lopezt-unity) [reply Editor/EditorCore/StripProBuilderScripts.cs]: I tried to put the scripts cleaning in a single method. 
However, each place using script stripping is doing this in a different way, which complicates this...

I also updated script stripping when exporting prefab and meshes.
They all call now the same method to ease future stripping extensions is needed.

File: Editor/MenuActions/Export/ExportAsset.cs [hunk 3]
```diff
@@ -98,11 +99,7 @@ public static string ExportWithFileDialog(IList<ProBuilderMesh> meshes, ExportAs
                         res = ExportMesh(assetPath, pb, options.replaceOriginal, true);
 
                         if (options.replaceOriginal)
-                        {
-                            pb.preserveMeshAssetOnDestroy = true;
-                            Undo.DestroyObjectImmediate(pb);
-                        }
-
+                            StripProBuilderScripts.DestroyProBuilderMeshAndDependencies(pb.gameObject, pb, true);
```
Comment (lopezt-unity) [inline Editor/MenuActions/Export/ExportAsset.cs:102]: @gytis-zasciurinskas All the modifications in this files come from a remark from @karl- .
Using Export>Export as mesh or prefab as causing error while performed on a gameobject with a ProBuilder Shape

**Review by karl- — APPROVED** [review]

**Review by gytis-zasciurinskas — APPROVED** [review]
We did a peer review with Thomas, the Export functionality now works correctly, no errors are now thrown
I1cxLubDPt

**Pull request merged**